attachments/views.py

```diff
@@ -18,99 +23,220 @@
-            file_uploaded.send(sender=f, request=request, session=session)
-            return JsonResponse({'ok': True, 'file_name': upload.file_name, 'file_size': upload.file_size}, content_type=content_type)
+            path = self.create_tmp_file()
+
+            self.validate_extension(path)
+            self.validate_max_size()
+            self.scan_clamd(path)
+
+            upload = self.create_upload(path)
+            self.set_session_data()
+
+            # Keeping the sender as 'self.file' for backwards compatibility
+            file_uploaded.send(sender=self.file, request=request, session=self.session)
+            return JsonResponse(
+                {'ok': True, 'file_name': upload.file_name, 'file_size': upload.file_size},
+                content_type=self.content_type,
+            )
 
         # These errors have helpful messages, so we return them
         except (InvalidExtensionException, InvalidFileTypeException, FileSizeException) as ex:
-            return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=content_type)
+            return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=self.content_type)
         except VirusFoundException as ex:
-            user = getattr(request, 'user', "Unknown user")
-            filename = upload.file_name
-            time_of_upload = datetime.now()
-            #if ATTACHMENTS_QUARANTINE_PATH is set, move the offending file to the quarantine, otherwise delete
-            attachments_quarantine_path = getattr(settings, 'ATTACHMENTS_QUARANTINE_PATH', None)
-            if attachments_quarantine_path:
-                quarantine_path = os.path.join(attachments_quarantine_path, os.path.basename(upload.file_path))
-                os.rename(upload.file_path, quarantine_path)
-                quarantine_msg = 'File has been quarantined here: {}'.format(quarantine_path)
-            else:
-                os.remove(upload.file_path)
-                quarantine_path = None
-                quarantine_msg = 'File has been removed from the system'
-
-            error_msg = force_text(ex)
-            log_message = "{} - Uploaded by {} at {}. {}.".format(error_msg,
-                                                                  user,
-                                                                  time_of_upload.strftime('%Y-%m-%d %H:%M:%S'), 
-                                                                  quarantine_msg)
-            logger.exception(log_message)
-
-            virus_detected.send(sender=upload, 
-                                user=user, 
-                                filename=filename, 
-                                exception=ex, 
-                                time_of_upload=time_of_upload, 
-                                quarantine_path=quarantine_path)
-            return JsonResponse({'ok': False, 'error': force_text(ex)}, content_type=content_type)
+            return handle_virus_found(request, upload)
```

Comment:
Looks like this should be `self.handle_virus_found(request, upload)`.

Reply:
Good catch! Pushed